internal/controller/datadogagent/override/global.go

```diff
@@ -435,3 +448,171 @@ func applyGlobalSettings(logger logr.Logger, manager feature.PodTemplateManagers
+		return err
+	}
+	dcaTokenResource(logger, dda, resourcesManager, podTemplateManager)
+	return nil
+}
+
+func dcaTokenDependencies(logger logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers) error {
+	global := dda.Spec.Global
+	var token string
+
+	// Prioritize existing secret
+	if isValidSecretConfig(global.ClusterAgentTokenSecret) {
+		return nil
+	}
+
+	// User specifies token
+	var key string
+	var hash string
+	var err error
+	if global.ClusterAgentToken != nil && *global.ClusterAgentToken != "" {
+		token = *global.ClusterAgentToken
+		// Generate hash
+		key = getDCATokenChecksumAnnotationKey()
+		hash, err = comparison.GenerateMD5ForSpec(map[string]string{common.DefaultTokenKey: token})
+		if err != nil {
+			logger.Error(err, "couldn't generate hash for Cluster Agent token hash")
+		} else {
+			logger.Info("built Cluster Agent token hash", "hash", hash)
+			// logger.V(2).Info("built Cluster Agent token hash", "hash", hash)
+		}
+	} else if dda.Status.ClusterAgent == nil || dda.Status.ClusterAgent.GeneratedToken == "" { // no token specified
+		token = apiutils.GenerateRandomString(32)
+	} else {
+		token = dda.Status.ClusterAgent.GeneratedToken // token already generated
+	}
+
+	// Create secret
+	secretName := secrets.GetDefaultDCATokenSecretName(dda)
+	if err := resourcesManager.SecretManager().AddSecret(dda.Namespace, secretName, common.DefaultTokenKey, token); err != nil {
+		return err
+	}
+
+	if key != "" && hash != "" {
+		// Add annotation to secret
+		if err := resourcesManager.SecretManager().AddAnnotations(logger, dda.Namespace, secretName, map[string]string{key: hash}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func dcaTokenResource(logger logr.Logger, dda *v2alpha1.DatadogAgent, resourcesManager feature.ResourceManagers, podTemplateManager feature.PodTemplateManagers) {
+	secretName := secrets.GetDefaultDCATokenSecretName(dda)
+	secretKey := common.DefaultTokenKey
+
+	global := dda.Spec.Global
+	if isValidSecretConfig(global.ClusterAgentTokenSecret) {
+		secretName = global.ClusterAgentTokenSecret.SecretName
+		secretKey = global.ClusterAgentTokenSecret.KeyName
+	}
+	// Add secret env var to pod
+	tokenEnvVar := common.BuildEnvVarFromSource(DDClusterAgentAuthToken, common.BuildEnvVarFromSecret(secretName, secretKey))
+	podTemplateManager.EnvVar().AddEnvVar(tokenEnvVar)
+
+	// Add annotation to pod template if secret has annotation
+	if obj, exists := resourcesManager.Store().Get(kubernetes.SecretsKind, dda.Namespace, secretName); exists {
+		key := getDCATokenChecksumAnnotationKey()
+		if val, ok := obj.GetAnnotations()[key]; ok {
+			podTemplateManager.Annotation().AddAnnotation(key, val)
+		}
+	}
+}
+
+func isValidSecretConfig(secretConfig *v2alpha1.SecretConfig) bool {
+	if secretConfig == nil {
+		return false
+	}
+	if secretConfig.SecretName == "" || secretConfig.KeyName == "" {
+		return false
+	}
+
+	return true
+}
```

Comment:
```suggestion
func isValidSecretConfig(secretConfig *v2alpha1.SecretConfig) bool {
	return secretConfig != nil && secretConfig.SecretName != "" && secretConfig.KeyName != ""
}
```
can rely on Go short circuit to directly return true/false: secretConfig is valid if and only if secretConfig is not nil AND secretConfig.SecretName is not empty AND KeyName is not empty.